pkg/provider/cloud/openstack/provider.go

```diff
@@ -289,22 +289,30 @@ func reconcileNetwork(ctx context.Context, netClient *gophercloud.ServiceClient,
 		}
 	}
 
-	// If NetworkName not specified, Create network with name kubernetes-clusterid
+	// This variable will be used to update the cluster object.
+	network := networkName
+
+	// If NetworkName not specified, Create network with name kubernetes-clusterid.
 	if networkName == "" {
+		network = resourceNamePrefix + cluster.Name
```

Comment:
I don't think adding another variable helps readability here.

I think your main issue stems from the fact that you're calling your own `createUserClusterNetwork` with a _network_ name, when the function in reality expects a _cluster_ name. Since whenever you want to create a network, you want to use the name determined by `reconcileNetwork`, you could just change your `createUserClusterNetwork` to be a `createNetwork(...networkName...)` and bam, problem solved.

Reply:
Okay, I was trying to avoid changing helper functions!